src/client/layout/layout_object.cpp

```diff
@@ -164,6 +166,73 @@ namespace client_layout
     return isScrollableInX || isScrollableInY;
   }
 
+  int LayoutObject::calculateLayer() const
+  {
+    auto parentPtr = parent();
+    if (parentPtr == nullptr)
+    {
+      // Root element has layer 0
+      return 0;
+    }
+    
+    // Start with parent's layer
+    int parentLayer = parentPtr->layer();
+    
+    // If parent is a scrollable container, children get parent's layer + 1
+    if (parentPtr->isScrollContainer())
```

Comment:
Check on the current node if it is scroll container, it means the layer number will be +1 on scrollable container only.